### ISSUE TITLE
--peer not respected for versions with ^ or ~

### ISSUE THREAD

**Issue by CreativeTechGuy** [raineorshine/npm-check-updates#1378]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14.14`

Note: I'm totally guessing the title of this issue. Please rename if you have a better idea of what the problem might be.

---

## Steps to Reproduce

Use the following `package.json`, run `npm install` and then `npm run test` and you'll see that it lists `esbuild` as an update that it'll try to do. (At the time of writing, the latest is `0.20.2`). But `@nx/esbuild` specifies it's `peerDependencies` as `"esbuild": "~0.19.2"` so that update should not be allowed.

```json
{
  "name": "ncu-bug",
  "scripts": {
    "test": "npm-check-updates --peer"
  },
  "devDependencies": {
    "@nx/esbuild": "^18.0.6",
    "esbuild": "^0.19.12",
    "npm-check-updates": "^16.14.16"
  }
}
```

**Comment by raineorshine** [comment]
Hi, thanks for reporting this.

Turns out this is a recurrence of #1181, which was not properly fixed. I had only fixed it for `>`, but any range operator like `^` or `@` should not be used in a spawned shell. The deeper issue is that the peer dependency lookup should use the exact version instead of the version declaration anyway.

I attempted a proper fix and published to `v16.14.17`. Let me know if that works.
